yt_dlp/extractor/nytimes.py

```diff
@@ -125,138 +91,427 @@ class NYTimesIE(NYTimesBaseIE):
+            'creator': creator,
+            'upload_date': unified_strdate(art_upload_date),
+            'timestamp': parse_iso8601(art_upload_date),
+            'thumbnails': thumbnails,
+        })
 
         return {
-            'id': remove_start(podcast_audio.get('target'), 'FT') or page_id,
-            'url': video_url,
-            'title': title,
-            'description': description,
-            'creator': track.get('credit'),
-            'series': podcast_title,
-            'episode': episode_title,
-            'episode_number': episode_number,
-            'duration': int_or_none(track.get('duration')),
+            **info
         }
 
+
+class NYTimesCookingIE(InfoExtractor):
+    IE_NAME = 'NYTimesCookingRecipes'
+    _VALID_URL = r'https?://cooking\.nytimes\.com/recipes/(?P<id>\d+)'
+    _TESTS = [{
+        'url': 'https://cooking.nytimes.com/recipes/1017817-cranberry-curd-tart',
+        'md5': '579e83bbe8e61e9de67f80edba8a78a8',
+        'info_dict': {
+            'id': '1017817',
+            'ext': 'mp4',
+            'title': 'Cranberry Curd Tart',
+            'description': 'md5:ad77a3fc321db636256d4343c5742152',
+            'timestamp': 1447804800,
+            'upload_date': '20151118',
+            'creator': 'David Tanis',
+            'thumbnail': r're:https?://\w+\.nyt.com/images/.*\.jpg',
+
+        },
+    }, {
+        'url': 'https://cooking.nytimes.com/recipes/1024781-neapolitan-checkerboard-cookies',
+        'md5': '58df35998241dcf0620e99e646331b42',
+        'info_dict': {
+            'id': '1024781',
+            'ext': 'mp4',
+            'title': 'Neapolitan Checkerboard Cookies',
+            'description': 'md5:ba12394c585ababea951cb6d2fcc6631',
+            'timestamp': 1701302400,
+            'upload_date': '20231130',
+            'creator': 'Sue Li',
+            'thumbnail': r're:https?://\w+\.nyt.com/images/.*\.jpg',
+        },
+    }, {
+        'url': 'https://cooking.nytimes.com/recipes/1019516-overnight-oats',
+        'md5': '2fe7965a3adc899913b8e25ada360823',
+        'info_dict': {
+            'id': '1019516',
+            'ext': 'mp4',
+            'timestamp': 1546387200,
+            'description': 'md5:8856ce10239161bd2596ac335b9f9bfb',
+            'upload_date': '20190102',
+            'title': 'Overnight Oats',
+            'creator': 'Genevieve Ko',
+            'thumbnail': r're:https?://\w+\.nyt.com/images/.*\.jpg',
+        }
+    }]
+
     def _real_extract(self, url):
         page_id = self._match_id(url)
-
         webpage = self._download_webpage(url, page_id)
 
-        video_id = self._search_regex(
-            r'data-videoid=["\'](\d+)', webpage, 'video id',
-            default=None, fatal=False)
-        if video_id is not None:
-            return self._extract_video_from_id(video_id)
+        json_obj = traverse_obj(
+            self._search_nextjs_data(webpage, page_id), ('props', 'pageProps'), default={})
```

Comment:
it's ok to make this fatal. `_search_nextjs_data` already is, and we have nothing to extract w/o this variable. I'd also use a var name like `next_data`
```suggestion
        next_data = self._search_nextjs_data(webpage, page_id)['props']['pageProps']
```